core/functions.py

```diff
@@ -36,46 +37,87 @@ def init(path=None, folder=None, config_settings:Dict=None, quiet=False, use_exi
-    with open(defaults, 'r') as f:
-        exec(compile(f.read(), defaults, 'exec'))
+    def __init__(self, path=None, folder=None, config_settings: Dict = None, quiet=False, otherJM=None, use_existing_folder=False):
+        self.otherJM = otherJM
 
-    config.update(config_settings or {})
+        if config.init:
+            return
 
-    from .jobmanager import JobManager
-    config.default_jobmanager = JobManager(config.jobmanager, path, folder, use_existing_folder)
+        if 'PLAMSDEFAULTS' in os.environ and isfile(expandvars('$PLAMSDEFAULTS')):
+            defaults = expandvars('$PLAMSDEFAULTS')
+        elif 'AMSHOME' in os.environ and isfile(opj(expandvars('$AMSHOME'), 'scripting', 'scm', 'plams', 'plams_defaults')):
+            defaults = opj(expandvars('$AMSHOME'), 'scripting', 'scm', 'plams', 'plams_defaults')
+        else:
+            defaults = opj(dirname(dirname(__file__)), 'plams_defaults')
+            if not isfile(defaults):
+                raise PlamsError('plams_defaults not found, please set PLAMSDEFAULTS or AMSHOME in your environment')
+        with open(defaults, 'r') as f:
+            exec(compile(f.read(), defaults, 'exec'))
 
-    if not quiet:
-        log('Running PLAMS located in {}'.format(dirname(dirname(__file__))), 5)
-        log('Using Python {}.{}.{} located in {}'.format(*sys.version_info[:3], sys.executable), 5)
-        log('PLAMS defaults were loaded from {}'.format(defaults), 5)
+        config.update(config_settings or {})
 
-        log('PLAMS environment initialized', 5)
-        log('PLAMS working folder: {}'.format(config.default_jobmanager.workdir), 1)
+        from .jobmanager import JobManager
+        config.default_jobmanager = JobManager(config.jobmanager, path, folder, use_existing_folder)
 
-    config.slurm = _init_slurm() if "SLURM_JOB_ID" in os.environ else None
+        if not quiet:
+            log('Running PLAMS located in {}'.format(dirname(dirname(__file__))), 5)
+            log('Using Python {}.{}.{} located in {}'.format(*sys.version_info[:3], sys.executable), 5)
+            log('PLAMS defaults were loaded from {}'.format(defaults), 5)
 
-    try:
-        import dill
-    except ImportError:
-        log('WARNING: importing dill package failed. Falling back to the default pickle module. Expect problems with pickling', 1)
+            log('PLAMS environment initialized', 5)
+            log('PLAMS working folder: {}'.format(config.default_jobmanager.workdir), 1)
+
+        config.slurm = _init_slurm() if "SLURM_JOB_ID" in os.environ else None
+
+        try:
+            import dill
+        except ImportError:
+            log('WARNING: importing dill package failed. Falling back to the default pickle module. Expect problems with pickling', 1)
+
+        config.init = True
 
-    config.init = True
+    def __enter__(self):
+        """Enter the context manager; return |init|."""
+        return self
```

Comment:
I know this is normally the default for context managers, but I think it is better to just return `None`. There is no usecase for having an instance of this object, I think it would only be a little confusing

Reply:
Hm, fine. The returning of an instance here was the initially the main motivator for also providing things like string representations and truthfulness, so there's little need to keep those around now (removed as well).